### PR TITLE
Create timeline view for UserFeed page

### PR DESCRIPTION
**Pull request by MonkeyDo**
See https://docs.google.com/document/d/1_rVY55eHksV0GuydG3J1Ooo7Ud_ypXeFGHnDOP1_On0

WIP: Waiting for API endpoints and view to be merged for the feed events. In the meantime I'm using a `fake-user-feed.json` with fake data. See PR #1318.

This PR also removes the temporary hardcoded feature flag ( a list of users allowed to view feed and follow buttons)

With the UserFeed component moved to its own directory in the tree, you will need to run `./develop.sh build static_builder` to refresh the static builder image.

preview

### CODE REVIEW

**Comment by paramsingh** [comment]
@MonkeyDo do we plan to show an event like "user x listened to this recording in their top 10 for the month" or something similar?

**Comment by MonkeyDo** [comment]
> @MonkeyDo do we plan to show an event like "user x listened to this recording in their top 10 for the month" or something similar?

Yes, probably. The event types are still a bit unclear to me, so I'd rather wait until we have a well defined for those event before implementing them.

The event types I've defined here I just invented, and I'm expecting they will need to be modified and extended:
https://github.com/metabrainz/listenbrainz-server/pull/1315/files#diff-5c646676854ba1bf2a7be197cb8880963677f273226692768a257d7f95ccae5dR26-R33

**Comment by shivam-kapila** [comment]
This looks good. Can we create a dir `timeline` under `src` and put the associated components there?

File: listenbrainz/webserver/static/js/src/APIService.test.ts [hunk 1]
```diff
@@ -619,6 +621,63 @@ describe("deleteListen", () => {
...
+      SyntaxError("Username missing")
+    );
+  });
+  it("throws appropriate error if userToken is missing", async () => {
+    await expect(apiService.getFeedForUser("Cthulhu", "")).rejects.toThrow(
+      SyntaxError("User token missing")
+    );
+  });
+
+  it("doesn't allow min_ts and max_ts at the same time", async () => {
```
Comment (paramsingh) [inline listenbrainz/webserver/static/js/src/APIService.test.ts:672]: this is not really necessary, the endpoint is ok with getting both min_ts and max_ts

File: listenbrainz/webserver/static/js/src/UserPageHeading.tsx [hunk 2]
```diff
@@ -48,17 +34,14 @@ const UserPageHeading = ({
   return (
     <h2 className="page-title">
       {user.name}
-      {loggedInUser &&
-        // TODO (param): Remove this when the feed feature is ready for release #feedfeatureflag
```
Comment (paramsingh) [inline listenbrainz/webserver/static/js/src/UserPageHeading.tsx:52]: 🎉

File: listenbrainz/webserver/static/js/src/types.d.ts [hunk 3]
```diff
@@ -453,6 +453,10 @@ declare type RecommendationFeedbackMap = {
   [recordingMbid: string]: RecommendationFeedBack | null;
 };
 
+/** ***********************************
```
Comment (paramsingh) [inline listenbrainz/webserver/static/js/src/types.d.ts:456]: this needs to be below the definition of `UserTrackRecommendationMetadata`

**Review by paramsingh — COMMENTED** [review]
Code looks good to me, just had a few comments.

File: listenbrainz/webserver/static/js/src/types.d.ts (hunk 3 above)
Comment (MonkeyDo) [reply listenbrainz/webserver/static/js/src/types.d.ts:456]: I sort of thought it was similar territory, but I'll move it.

**Review by paramsingh — APPROVED** [review]
🚢

File: consul_config.py.ctmpl [hunk 4]
```diff
@@ -171,3 +171,8 @@ WEBSOCKETS_SERVER_URL = '''{{template "KEY" "websockets_server_url"}}'''
 SESSION_REMEMBER_ME_DURATION = 365
 
 LISTEN_DUMP_TEMP_DIR_ROOT = '''{{template "KEY" "listen_dump_temp_dir"}}'''
+
+# Feature flag: only enable follow button in beta and test
+{{if not eq (env "DEPLOY_ENV") "prod"}}
+FEATURE_FOLLOW_BUTTON = True
+{{end}}
```
Comment (MonkeyDo) [inline consul_config.py.ctmpl:178]: Your thoughts on this @paramsingh ?
I haven't been able to test it.

Do you have a way to test this locally?

 Maybe we can try on test.lb first?
We might need to do the db migration on the test db if so.

**Review by paramsingh — APPROVED** [review]

**Pull request merged**